### ISSUE TITLE
No Weed after Gamecrash or Serverrestart

### ISSUE THREAD

**Issue by HitzefreiTV** [DemmyDemon/esx_uteknark#30]
Hey,

first of all, thank you for making your resource available to us for free. I really like it.

However i noticed one thing. Is it normal that all plants disappear when I restart my game or the server?

I look forward to your response.

Best regards

HitzefreiTV

**Comment by DemmyDemon** [comment]
Yes, this is entirely expected if the database is not set up right.
If you have set it up right, then it's some other issue, but I have no way of guessing what the problem might be because you have not provided any logs or anything.

Basically,
```sql
CREATE TABLE `uteknark` (
    `id` INT(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
    `stage` INT(3) UNSIGNED NOT NULL DEFAULT 1,
    `time` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
    `x` FLOAT(10) NOT NULL,
    `y` FLOAT(10) NOT NULL,
    `z` FLOAT(10) NOT NULL,
    `soil` BIGINT(20) NOT NULL,
    INDEX (`stage`, `time`)
);

**Comment by HitzefreiTV** [comment]
![image](https://user-images.githubusercontent.com/106077626/169833614-08095b35-8314-43e9-b3a6-20efe55ba6ea.png)
![image](https://user-images.githubusercontent.com/106077626/169833656-991e760a-de5d-40d2-b608-04dbb7782bd3.png)

Thank you for your prompt reply.
Here you can see the structure of the table and that values are also entered in the table.

What logs do you need to isolate the error? We haven't changed anything in the script. Console recordings, chat messages, DB messages?

Best regards,

HitzefreiTV

**Comment by DemmyDemon** [comment]
All the server log lines with `<esx_uteknark>` in them would be a great start.

Are you absolutely sure that this is from the same database? Those plants were planted in March, and would be long dead or harvested by now. The fact that they're still there in the database tells me there is something else afoot.

Well, unless the clock on your server is running slow by _two months_, I guess ;-)

**Comment by HitzefreiTV** [comment]
Hi,

when i restart my Server, this is the result:

![image](https://user-images.githubusercontent.com/106077626/169842730-43609f70-a309-4351-a6d3-b6524a5884af.png)

**Comment by HitzefreiTV** [comment]
![image](https://user-images.githubusercontent.com/106077626/169843752-96ba0398-e647-4499-ac37-a0b936d8598f.png)
And this is line 46 in cropstate.lua

**Comment by DemmyDemon** [comment]
Yep, that's because one or more of the plants are outside the octree bounds.

More area has been added to the map since I made this, so the octree doesn't cover the whole map.

In [lib/cropstate.lua, on line 137](https://github.com/DemmyDemon/esx_uteknark/blob/master/lib/cropstate.lua#L137), assuming you have an unmodified version, bump up the size of the octree (second vector3 parameter).

Note that this will likely move all active plants.

Best of luck.
